### ISSUE TITLE
Menu won't show if I segue to another VC

### ISSUE THREAD

**Issue by EduardoGin** [kukushi/SideMenu#52]
Hello,

So, I have imported this pod, tested the example and everything worked perfect.
I can set it so it shows the menu on the initial screen.

The Menu is also persistent if I choose cell 0 (which is the initial cell). Meaning if I choose that cell it will stay on my initial VC and the menu keeps appearing/

Now the problem is: If I segue from the buttons on my VC, the menu won't appear anymore, even though I have on my code `  sideMenuController?.cache(viewControllerGenerator: {
            self.storyboard?.instantiateViewController(withIdentifier: "StandardVC")
        }, with: "1")`

Am I missing something obvious @kukushi ?
Here is a Print Screen of my storyboard, thanks :)

P.S.: the VC with the 3 big buttons (standard, isa, pension) is the initial VC
![Screenshot 2019-04-15 at 11 57 45](https://user-images.githubusercontent.com/47526850/56127720-b5f42700-5f75-11e9-8b78-d1e6aa1670cc.png)

**Comment by kukushi** [comment]
@EduardoGin The root view controller need to be the side menu controller.

**Comment by EduardoGin** [comment]
Hi @kukushi , I will give it a look in 30min. I will update you, thanks

**Comment by EduardoGin** [comment]
Any ideas @kukushi ? :)

**Comment by kukushi** [comment]
@EduardoGin SideMenu should be the root view controller and top level view controller, which is parent of all other view controllers.

If you want to add the login screen, I will suggest you changing the root view controller to side menu after successfully login.

**Comment by EduardoGin** [comment]
Hi @kukushi thanks for replying 

My SideMenu is already the root viewcontroller as you can see from the screenshot, the arrow is there, making it the FIRST VIEW, therefore the root view controller.

The login screen I have that sorted, I just don't get how it is still not working when it is the root viewcontroller.

Like I said, everything works with your Pod. It is an excellent pod!
The only problem I have is when I segue. The menu doesn't show.

Unfortunately there are no segues in your Example, so I couldn't take anything from there, so I am not sure what the problem is.

TL;DR: All I want is to able to segue and the menu still show. Any ideias how this is achievable with your library @kukushi  :)

Thanks

**Comment by kukushi** [comment]
I’m afraid there is some information missing here.
Can you send me a demo for reproducing this problem?

Xing He,
Best Regards

**Comment by EduardoGin** [comment]
I am sure there is...

I will put things by bullet points

1. Menu is working as it should.
2. SideMenu is the RootVC
3. If I segue to another VC using the menu, it segues properly
3.1 If I try to open the Menu on the new VC, the menu opens
4. If I segue to another VC (example: button click), the menu won't appear on the VC

Here is a link with the Zip. Open the .workspace
https://drive.google.com/open?id=1_LSiZxgZdWHQ2LNUmUyES6KUwswfoJYc

Some notes:

. Just click Login. (It's a  normal segue)
. PortfolioDetailsVC is the first View after the Login
. On PortfolioDetailsVC, click the "Standard" Button, it will segue you to the next View: "StandardVC"

Thank @kukushi

**Comment by kukushi** [comment]
@EduardoGin Thanks for your code.

There are some misuse in your code:
1. Side menu should have only one content segue, not two.
2. I will suggest you using the navigation controller instead of "Push detail"

**Comment by EduardoGin** [comment]
Hi @kukushi I appreciate the time you took to look at it.

I have some questions:

My app is very navigation bar oriented, meaning I will have to create one navigation bar nearly for every VC so I can go back and forth, that's why I created it manually, to avoid spamming my Main story with navigation controller VCs.

My questions are:

1. If I create a navigation bar, I will have to have more than 1 content segue, since the whole app is based on it (navigation wise), right? One content segue per Navigation bar.

2. Is there a way I can use it designed the way I have?

Thanks

**Comment by kukushi** [comment]
@EduardoGin 
Maybe you have misunderstand how Navigation Controller work. The whole navigation stack share one single navigation bar. You can follow the way example use `pushViewController`.

**Comment by EduardoGin** [comment]
Hi @kukushi , yeah my bad.

One question, is it possible to expand the navigation controller so it is as big as the one I have?
